v0/destinations/profitwell/transform.js

```diff
@@ -0,0 +1,294 @@
+      if (subscriptionId) {
+        throw new CustomError("profitwell subscription_id not found", 400);
+      }
+      payload = constructPayload(message, createPayloadMapping);
+      payload = {
+        ...payload,
+        user_id: userId,
+        user_alias: userAlias
+      };
+      if (
+        payload.plan_interval &&
+        !(
+          payload.plan_interval.toLowerCase() === "month" ||
+          payload.plan_interval.toLowerCase() === "year"
+        )
+      ) {
+        throw new CustomError("invalid format for planInterval. Aborting", 400);
+      }
+      if (
+        payload.plan_currency &&
+        !isValidPlanCurrency(payload.plan_currency)
+      ) {
+        payload.plan_currency = null;
+      }
+      if (
+        payload.status &&
+        !(
+          payload.status.toLowerCase() === "active" ||
+          payload.status.toLowerCase() === "trialing"
+        )
+      ) {
+        payload.status = null;
+      }
+      payload.effective_date = unixTimestampOrError(
+        payload.effective_date,
+        message.originalTimestamp
+      );
+      response.method = defaultPostRequestConfig.requestMethod;
+      response.endpoint = `${BASE_ENDPOINT}/v2/subscriptions/`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      return response;
+    }
+
+    // for a given userId, subscription is found at dest.
+    if (valFound) {
+      payload = constructPayload(message, updatePayloadMapping);
+      if (
+        payload.plan_interval &&
+        !(
+          payload.plan_interval.toLowerCase() === "month" ||
+          payload.plan_interval.toLowerCase() === "year"
+        )
+      ) {
+        throw new CustomError("invalid format for planInterval. Aborting", 400);
+      }
+      if (
+        payload.status &&
+        !(
+          payload.status.toLowerCase() === "active" ||
+          payload.status.toLowerCase() === "trialing"
+        )
+      ) {
+        payload.status = null;
+      }
+      payload.effective_date = unixTimestampOrError(
+        payload.effective_date,
+        message.originalTimestamp
+      );
+      response.method = defaultPutRequestConfig.requestMethod;
+      response.endpoint = `${BASE_ENDPOINT}/v2/subscriptions/${subscriptionId ||
+        subscriptionAlias}/`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      return response;
+    }
+  }
+
+  // handler for other destination side errors
+  if (res.response.response.status !== 404) {
+    throw new CustomError(
+      "Failed to get subscription history for a user",
+      res.response.response.status
+    );
+  }
+
+  // drop event if profitwellUserId (externalId) did not match with any user_id
+  if (userId) {
```

Comment:
this looks a bit confusing. We are checking here that `userId` has a valid value but was not found on fetch call, hence throwing the error. Is that correct?

Reply:
Yes, if `userId` is provided then it means we are trying to update the subscription for that particular user after fetching it. `userId` is automatically generated by ProfitWell for any new users.

Reply:
cool